CRM/Contribute/BAO/Contribution.php

```diff
@@ -2755,6 +2755,10 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
         $balanceTrxnParams['status_id'] = $statusId;
         $balanceTrxnParams['payment_instrument_id'] = $params['contribution']->payment_instrument_id;
         $balanceTrxnParams['check_number'] = CRM_Utils_Array::value('check_number', $params);
+        if (!empty($balanceTrxnParams['from_financial_account_id']) && 
+            ($statusId == array_search('Completed', $contributionStatuses) || $statusId == array_search('Partially paid', $contributionStatuses))) {
+          $balanceTrxnParams['is_payment'] = 1;
+        }
```

Comment:
Is payment flag is set to 1 in both cases, partial and complete. It's just that the code which handles setting the is_payment for complete transactions is handled elsewhere.